### DISCUSSION TITLE
License for Kafka/Protobuf support

### DISCUSSION THREAD

**Discussion by xdgrulez** [trinodb/trino#22421] [Ideas]
Hi,

looking at the Trino code (kafka plugin), I see that Protobuf support is only optional (you have to bring your own JAR) seemingly because the Protobuf client support for Confluent Schema Registry is using the Confluent Community License but not Apache.

This is wrong. When you look at the GitHub repo for Confluent's Schema Registry, the LICENSE file states the following:

_The project is licensed under the Confluent Community License, except for client
libs, which is under the Apache 2.0 license._

Hence, I do not see a reason why not to natively support Protobuf (and also JSONSchema, BTW, I just started another discussion on this one) in Trino as long as Confluent does not change their license model (which is highely unlikely in this case).

Is anybody responsible for the kafka plugin? Or should I just create a pull request for that and see who comes up? ;-)

Best, Ralph

**Comment by hashhar** [comment]
I assume you're referring to this sentence? https://github.com/confluentinc/schema-registry/tree/master?tab=readme-ov-file#license?

cc: @mosabua can you please help confirm this?

**Comment by xdgrulez** [comment]
Hi, I referred to the LICENSE file:




*The project is licensed under the Confluent Community License, except for
clientlibs, which is under the Apache 2.0 license.See LICENSE file in each
subfolder for detailed license agreement.*

But you might well be right - in the README.md, it says only the avro- and
client- libs are under the Apache license, that would mean that the
protobuf- and jsonschema- really are under Confluent Community...

Best,
Ralph


Am Di., 25. Juni 2024 um 12:27 Uhr schrieb Ashhar Hasan <
***@***.***>:

**Comment by mosabua** [comment]
I saw that it says the license is specific for different folders and indeed there are license files in various directories, for example:

* https://github.com/confluentinc/schema-registry/blob/master/json-serializer/LICENSE
* https://github.com/confluentinc/schema-registry/blob/master/json-schema-serde/LICENSE

So JSON might be fine .. but protobuf there are no license files .. so its unclear. If we want to improve this we have to figure license out for each specific dependency we need (and the transitive dependencies of it). Currently it doesnt look very clear to me .. so we might have to ask upstream for clarification depending on what we find.

Also note that realistically the applicable license should be detailed in the pom.xml file but that is not the case from all I can tell. 

See https://github.com/confluentinc/schema-registry/blob/master/json-serializer/pom.xml

**Comment by mosabua** [comment]
In a nutshell.. feel free to investigate and send a PR with your findings and then we can review and check for next steps. Seems like JSON might be easier to determine than protobuf so splitting the work might help.

**Comment by xdgrulez** [comment]
Hi Manfred,

I'll ask Confluent support about this and post their response here :)

Best, Ralph

Am Di., 25. Juni 2024 um 17:05 Uhr schrieb Manfred Moser <
***@***.***>:

**Comment by xdgrulez** [comment]
Hi Manfred,

I got the response from Confluent finally.

The project is licensed under the Confluent Community License, except for
the client-* and avro-* libs, which are under the Apache 2.0 license. See
LICENSE file in each subfolder for detailed license agreement. In general,
client-side libraries are Apache and server-side libraries are CCL.



However, we kept most of the protobuf-* and json-schema-* libraries as CCL
if they have meaningful code, especially the compatibility checkers.

So the assumption that the Protobuf support should not be directly included
in Trino is correct as it stands, unfortunately.

Best, Ralph

Am Di., 25. Juni 2024 um 17:24 Uhr schrieb Ralph Matthias Debusmann <
***@***.***>:

**Reply by hashhar** [reply]
Thanks for your time to follow up on this, unfortunate answer, but it's good to have clarity regarding this.

**Reply by mosabua** [reply]
Yes.. another proof that while Apache Kafka is open source, Confluent and Confluent Community License libraries are clearly not open source.

I think that means we can close this issue. Again .. sorry but there is nothing we can do beyond this.
